### PR TITLE
fix(ci): secure code-scan action releases

### PR DESCRIPTION
**Pull request by mldangelo-oai**
Split code-scan mirror releases into build, attest, and publish jobs. Build without write credentials, require successful attestation before publishing the complete payload, and reject commits containing unexpected mirror paths. Constrain the documented verification command to the release workflow and main branch.

Remove the scanner's temporary install after success or failure, without masking the scan result when cleanup fails. Preserve main's isolated installation and action-Node launcher; simplify the install comments.

## Validation

- 89 action and release-workflow tests passed, along with action/root TypeScript checks, the action build, actionlint, lint, and formatting.
- Executed the production scan function with a local npm fixture and real scanner subprocesses. Paths with spaces and literal shell characters were preserved; success, install failure, and scanner failure all removed the temporary install. GitHub/npm tokens were excluded from the relevant child environments. No package or mirror release was published.

The publication checkout also disables Git hooks. A real temporary Git repository with a post-checkout hook reproduces the prior execution and verifies the workflow command suppresses it. The 89 focused tests, TypeScript, actionlint, lint, and formatting pass.

### CODE REVIEW

**Review by jameshiester-oai — COMMENTED** [review]
No actionable findings.

I reviewed the release workflow sequencing, credential boundaries, mirrored-path allowlist, attestation payload, action-owned CLI prefix, direct executable invocation, cleanup behavior, tests, and user-facing supply-chain documentation. The build-attest-publish split is coherent: the foreign-repo token is minted only after the payload is built and attested, and mirror publication is constrained to the intended generated paths. I did not find a regression, unintended behavior change, documentation mismatch, or worthwhile simplification that should block this head.

Validation:
- Node 24.18: focused Vitest passed (2 files, 62 tests).
- code-scan-action TypeScript checks passed after installing its local dependencies.
- git diff --check passed.
- Current GitHub checks are green.

No inline comments.

The requested full PR-review command was unavailable on this host, so I applied the manual full-review lenses: correctness, regression/edge-case behavior, tests, docs, error handling, API/types, and simplification.

**Comment by mldangelo-oai** [comment]
@codex review

File: code-scan-action/src/main.ts [hunk 1]
```diff
@@ -412,52 +417,66 @@ async function installPromptfooCli(promptfooVersion: string): Promise<void> {
...
+  try {
+    const promptfooCli = await installPromptfooCli(promptfooVersion, npmrcDir);
+
+    core.info('🚀 Running promptfoo code-scans run...');
+
+    let scanOutput = '';
+    let scanError = '';
+    const scanEnv = createScanEnv(oidcToken);
+
+    const exitCode = await exec.exec(`"${promptfooCli.replaceAll('"', '\\"')}"`, cliArgs, {
```
[the remark this answers is not in the document]
Comment (mldangelo-oai) [reply code-scan-action/src/main.ts]: The installed @actions/exec implementation parses commandLine with argStringToArray before invoking ToolRunner, so quoted executable paths are supported. After syncing with main, this branch retains the action Node launcher and passes the installed scanner path in the argument array. The production scan function passed a real subprocess fixture under a path containing spaces on 9f7f342b4541890c7854e2547fec97d9f942ac33.

**Comment by mldangelo-oai** [comment]
@codex review

Please review the current head 9f7f342b4541890c7854e2547fec97d9f942ac33.

File: .github/workflows/release-please.yml [hunk 2]
```diff
@@ -320,6 +309,67 @@ jobs:
...
+      # Mint the foreign-repository write token only after all dependency
+      # installation, build, and payload preparation steps have completed.
+      - name: Create app token for code-scan-action mirror
+        uses: actions/create-github-app-token@bcd2ba49218906704ab6c1aa796996da409d3eb1 # v3
+        id: mirror-token
+        with:
+          app-id: ${{ vars.PROMPTFOOBOT_APP_ID }}
+          private-key: ${{ secrets.PROMPTFOOBOT_APP_PRIVATE_KEY }}
+          owner: promptfoo
+          repositories: code-scan-action
```
[the remark this answers is not in the document]
Comment (mldangelo-oai) [reply .github/workflows/release-please.yml:372]: Fixed in 7739d4a2fbef8ba0ba41d2a49bce7ebdc9a836b6. The mirror checkout now uses the same hooksPath protection as clone, commit, and push. A real Git hook fixture failed before the fix and passes with the workflow command; all 89 focused action/release tests and required local checks pass.

**Comment by mldangelo-oai** [comment]
@codex review

Please review current head `7739d4a2fbef8ba0ba41d2a49bce7ebdc9a836b6`.

**Review by jameshiester-oai — COMMENTED** [review]
No new actionable findings at this head. The earlier credentialed checkout hook risk is addressed by disabling hooks on the branch switch; I checked the token boundary, artifact attestation gate, mirror path allowlist, action CLI install and cleanup paths, and the existing review threads. Focused code-scan action and release tests passed (80/80), including the real Git hook fixture; 44 reported checks are green. The /pr-review-toolkit:review-pr command was unavailable (exit 127), so I performed a bounded manual security, correctness, API, concurrency, tests, and maintainability review.\n\n
